src/TaskSerializer/index.ts

```diff
@@ -0,0 +1,57 @@
+import type { Task } from '../Task';
+
+type Writeable<T> = { -readonly [P in keyof T]: T[P] };
+
+/**
+ * A subset of fields of {@link Task} that can be parsed from the textual
+ * description of that Task.
+ *
+ * All fields are writeable for convenience.
+ */
+export type TaskDetails = Writeable<
+    Pick<
+        Task,
+        | 'description'
+        | 'priority'
+        | 'startDate'
+        | 'createdDate'
+        | 'scheduledDate'
+        | 'dueDate'
+        | 'doneDate'
+        | 'recurrence'
+        | 'tags'
+    >
+>;
+
+/**
+ * An abstraction that manages how a {@link Task} is read from and written
+ * to a file.
+ *
+ * A {@link TaskSerializer} is only responsible for the single line of text that follows
+ * after the checkbox:
+ *
+ *        - [ ] This is a task description
+ *              ~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * @exports
+ * @interface TaskSerializer
+ */
+export interface TaskSerializer {
+    /**
+     * Parses task details from the string representation of a task
+     *
+     * @param line The single line of text to parse
+     * @returns The parsed {@link TaskDetails}, or null the string could not be parsed
```

Comment:
I think I might tweak the wording here:

```diff
- or null the string could not be parsed
+ or null of nothing at all could be parsed from the string
```

Alternatively, the previous behaviour was that anything that could not be parsed was included in the description, unchanged - and maybe that is a safer behaviour here?

Trying to express why I am nervous about this: The behaviour of returning `null` will mean that users will get no feedback, ever, if one of their intended task lines cannot be parsed. The task will be invisible, never searched, never displayed in search results. 

They will also not be able to search for their invalid tasks.

Reply:
I am very much open to being persuaded that `null` is a valid thing to return.... I don't know what I don't know about possible future intended behaviours!